### DISCUSSION TITLE
Is there instructions on how add new program type?

### DISCUSSION THREAD

**Discussion by MajdAlhayek** [DustinBrett/daedalOS#247] [Feature Requests]
First of all, thanks a lot for this great tool! 

I am trying to create a way to emulate "Outlook" like experience where the user can option outlook and send an email. The email won't need to be actually sent, but I would fire a custom Javascript event instead like "CustomEmailSent" with the info from outlook.

Is there a how-to documents that explain what is needed in order to add this the daedalOS? The idea is to do something like the screenshots below

Also, is there a way to run the emulator inside an html div of a current page instead of a stand alone app?

![image](https://user-images.githubusercontent.com/110922525/208500076-57118ec8-202e-454c-8f17-dea869b11f72.png)

![image](https://user-images.githubusercontent.com/110922525/208500123-32c4c4c9-75c7-4601-b3b4-501d44992da1.png)

**Comment by DustinBrett** [comment]
Thanks very much and this is a great idea! I actually do plan to make an Email app in the near future that will fully function. The only thing stopping me is trying to find a way to do it without having to do anything on the backend myself. Ideally using a email sending api. I also plan for this app to be able to save the sent emails and to open eml files using something like https://github.com/nodemailer/mailparser

I will put this feature as high priority. Until then there is no way to make custom apps without working in the source code yourself.

**Comment by MajdAlhayek** [comment]
@DustinBrett interaction with a email will be great! I am guessing there would be a way to configure the email sender settings somehow is the app itself.

Any idea how long it'll take to add Outlook to the app? Anything I can do to help expedite this?

**Reply by DustinBrett** [reply]
It will probably take months for me to sit down and finish it to the level I want to commit it. When I do it may indeed allow changing the `To` address. If you are confident in being able to build an app to the level of the other apps I have, you are welcome to make it to expedite things. It would possibly have a similar structure to the PDF app (https://github.com/DustinBrett/daedalOS/tree/main/components/apps/PDF) and ideally will look like whatever email client comes with Windows 10. I've used Gmail for ages so have not kept up with how Outlook looks.

**Reply by MajdAlhayek** [reply]
@DustinBrett were you ever able to get to this and implement an Outlook or any other emailing app?

**Reply by DustinBrett** [reply]
Hey @MajdAlhayek,

Sorry to say I never did get a solution to this yet. I am thinking in the near future I will start to add apps and features which require API keys to work, and in that case I can make use of existing API's which require keys to work. But until then I have not yet made a mail app.
